src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -172,30 +195,41 @@ public partial class DistributedApplicationExecutionContext
     {
         public DistributedApplicationExecutionContext(DistributedApplicationExecutionContextOptions options) { }
 
+        public DistributedApplicationExecutionContext(DistributedApplicationOperation operation, string publisherName) { }
+
         public DistributedApplicationExecutionContext(DistributedApplicationOperation operation) { }
 
+        public bool IsInspectMode { get { throw null; } }
+
         public bool IsPublishMode { get { throw null; } }
 
         public bool IsRunMode { get { throw null; } }
 
         public DistributedApplicationOperation Operation { get { throw null; } }
 
+        public string PublisherName { get { throw null; } set { } }
+
         public System.IServiceProvider ServiceProvider { get { throw null; } }
     }
 
     public partial class DistributedApplicationExecutionContextOptions
     {
+        public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation, string publisherName) { }
+
         public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation) { }
 
         public DistributedApplicationOperation Operation { get { throw null; } }
 
+        public string? PublisherName { get { throw null; } }
+
         public System.IServiceProvider? ServiceProvider { get { throw null; } set { } }
     }
 
     public enum DistributedApplicationOperation
     {
         Run = 0,
-        Publish = 1
+        Publish = 1,
+        Inspect = 2
```

Comment:
Should we make this experimental? @mitchdenny

Reply:
@mitchdenny do we need this for 9.2? if so can you do it today?

Reply:
https://github.com/dotnet/aspire/pull/8665